design/overview.md

```diff
@@ -26,48 +26,49 @@ The first two steps are done once, the other 3 repeated many times.
+- `MultiSig Set` - The set of Ethereum keys along with respective voting power. This is based on the validator set and mapped over the registered keys. However, as this is a different chain, this is mirrored with a delay.
+- `Peggy Bridge Tx pool` - Is a transaction pool that exists in the chain store of Cosmos -> Ethereum transactions waiting to be placed into a transaction batch
+- `Transaction batch` - A transaction batch is a set of Ethereum transactions to be sent from the Peggy Ethereum contract at the same time. This helps reduce the costs of submitting a batch. Batches have a maximum size (currently around 100 transactions) and are only involved in the Cosmos -> Ethereum flow
+- `Peggy Bridge Batch pool` - Is a transaction pool like strucutre that exists in the chains to store, seperate from the `Peggy Bridge Tx pool` it stores transactions that have been placed in batches that are in the process of being signed or being submitted by the `Orchestrator Set`
+- `EthBlockDelay` - Is a agreed upon number of Ethereum blocks all oracle attestations are delayed by. No `Orchestrator` will attest to have seen an event occur on Ethereum until this number of blocks has elapsed as denoted by their trusted Ethereum full node. This should prevent short forks form causing disagreements on the Cosmos side. The current value being consdiered is 50 blocks.
+- `Last ValSet` - The `Last Valset` is the last validator set known to have been accepted by the Ethereum chain. When a new ValSet update is submitted it is first `Confirmed` making it possible to submit and then `Observed` updating the value of `Last ValSet` in the store.
+- `Confirmed` - Many actions in Peggy require RLP encoded signatures to be submitted by the `Orchestrators`, an operation is `Confirmed` when it is possible to execute it on Ethereum by collecting and submitting these RLP encoded signatures representing 66% of all voting power encoded in the `Last Valset`.
+- `Observed` - events on Ethereum are considered `Observed` when 66% of the active Cosmos validator set during a given block has submitted an oracle message attesting to seeing the event. Note this is distinct from `Confirmed`! At any given time the set of who can `Observe` an event and who can `Confirm` and event will be slightly different. Since `Confirmed` is based off of the `Last ValSet` and `Observed` is always up to date with the latest validator set.
+- `Validator set delta` - This is a term for the difference between the validator set currently in the Peggy Ethereum contract and the actual validator set on the Cosmos chain. Since the validator set may change every single block there is essentially guaranteed to be some nonzero `Validator set delta` at any given time.
+- `Allowed validator set delta` - This is the maximum allowed `Validator set delta` this parameter is used to determine if a Peggy contract should be adopted and if a validator set update should be sent.
```

Comment:
`Allowed validator set delta` is an important concept below. Is this set _subjectively_ (by each Orchestrator), or _objectively_ (by an on-chain parameter)?

Reply:
"if a Peggy contract should be adopted" means 'the address goes into the store as the official bridge address and everything conditioned on that moves forward' 

Currently the way I've (not very clearly) dealt with parameters is that when someone proposes a contract with MsgProposeContract (lets just call it that) that message contains a number of parameters as well. So you're not only agreeing to the contract but also the included params.

Obviously this could lead to someone submitting a valid contract and not having it adopted because they also wish to adopt parameters that no one wants. I think the general idea is that validators will need to select shelling point values for these parameters and agree on them off chain, because coding tie breaking votes for every individiaul parameter seems difficult. Too much optionality even now with just two, what if we add more?